public/app/core/AppWrapper.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import { Router, Route, Redirect, Switch } from 'react-router-dom';
+import angular from 'angular';
+import { each, extend } from 'lodash';
+import { config, getLocationService } from '@grafana/runtime';
+import { Provider } from 'react-redux';
+import { store } from 'app/store/store';
+import { ErrorBoundaryAlert, ModalRoot, ModalsProvider } from '@grafana/ui';
+import { GrafanaApp } from '../app';
+import { routes } from 'app/routes/routes';
+import { ConfigContext, ThemeProvider } from './utils/ConfigProvider';
+import { RouteDescriptor } from './navigation/types';
+import { contextSrv } from './services/context_srv';
+import { SideMenu } from './components/sidemenu/SideMenu';
+import { navigationLogger, queryStringToJSON, shouldForceReload } from './navigation/utils';
+import { updateLocation } from './actions';
+import { GrafanaRoute, SyncLocationWithRedux } from './navigation/GrafanaRoute';
+
+interface AppWrapperProps {
+  app: GrafanaApp;
+}
+
+interface AppWrapperState {
+  ngInjector: any;
+}
+
+export class AppWrapper extends React.Component<AppWrapperProps, AppWrapperState> {
+  container = React.createRef<HTMLDivElement>();
+
+  constructor(props: AppWrapperProps) {
+    super(props);
+
+    this.state = {
+      ngInjector: null,
+    };
+  }
+
+  componentDidMount() {
+    if (this.container) {
+      this.bootstrapNgApp();
+    } else {
+      throw new Error('Failed to boot angular app, no container to attach to');
+    }
+  }
+
+  bootstrapNgApp() {
+    const { app } = this.props;
+    const invoker = angular.bootstrap(document, app.ngModuleDependencies);
+    navigationLogger('AppWrapper', false, 'Angular app bootstrap');
+    this.setState(
+      { ngInjector: invoker },
+      invoker.invoke(() => {
+        each(app.preBootModules, (module) => {
+          extend(module, app.registerFunctions);
+        });
+        app.preBootModules = null;
+        // I don't know
+        return () => {};
+      })
+    );
+  }
+
+  renderRoute = (route: RouteDescriptor) => {
+    // const { updateLocation } = this.props;
+    // TODO[Router]
+    // @ts-ignore
+    const roles = route.roles ? route.roles() : [];
+
+    return (
+      <Route
+        exact
+        path={route.path}
+        key={`${route.path}`}
+        render={(props) => {
+          navigationLogger('AppWrapper', false, 'Rendering route', route, 'with match', props.location);
+          // TODO[Router]: test this logic
+          if (roles && roles.length) {
+            if (!roles.some((r: string) => contextSrv.hasRole(r))) {
+              return <Redirect to="/" />;
+            }
+          }
+
+          return (
+            <SyncLocationWithRedux {...props}>
+              <GrafanaRoute
+                {...props}
+                component={route.component}
+                route={route}
+                $injector={this.state.ngInjector}
+                $contextSrv={contextSrv}
```

Comment:
we do not need this, it's a public singleton and not an angular service (it is but that is only for old angular components)

Reply:
yes, it's on TODOs to remove